ouroboros-consensus/ouroboros-consensus-test-infra/src/Test/Util/Golden.hs

```diff
@@ -0,0 +1,96 @@
+module Test.Util.Golden
+  ( goldenTestCBOR
+  ) where
+
+import           Codec.CBOR.Encoding (Encoding)
+import           Codec.CBOR.Write (toStrictByteString)
+import qualified Data.ByteString as Strict
+import qualified Data.ByteString.Base16 as B16
+import           Data.List (intercalate, uncons)
+import           Data.List.NonEmpty (NonEmpty (..))
+import qualified Data.List.NonEmpty as NE
+import           Data.Tuple (swap)
+
+import           Ouroboros.Consensus.Util (chunks)
+
+import           Test.Tasty.HUnit
+
+-- | Test whether the encoding of the given example value matches the expected
+-- binary encoding, given as a hexadecimal bytestring.
+--
+-- Hint: when writing the test, just pass an empty expected string.
```

Comment:
Be still my beating heart! Thank you for anticipating and accomodating this need.